tests/console/ping.pm

```diff
@@ -35,13 +34,23 @@ sub run {
     my $ifname = script_output('ip -6 link |grep "^[0-9]:" |grep -v lo: | head -1 | awk "{print \$2}" | sed s/://');
     my $addr = script_output("ip -6 addr show $ifname | grep 'scope link' | head -1 | awk '{ print \$2 }' | cut -d/ -f1");
 
-    my $cmd = "ping6 -c2 $addr%$ifname";
+    # test as non-root user
+    my $sudo = 'sudo -u \#1000';
+    record_info('id non-root', script_output("$sudo id", proceed_on_failure => 1));
+
+    my $cmd = "$sudo ping -c2 ::1";
+    record_info('ping6 ::1');
+    my $rc = script_run($cmd);
+    $self->result("fail") if ($rc);
+
+    $cmd = "$sudo ping6 -c2 $addr%$ifname";
     record_info('ping %');
-    assert_script_run($cmd);
+    $rc = script_run($cmd);
+    $self->result("fail") if ($rc);
```

Comment:
why not `assert_script_run` to handle return value?

Reply:
+1, used for all but the last commit which needs evaluation for soft failure.